snapshot/snapshot.go

```diff
@@ -200,6 +212,8 @@ func NewSnapshotter(ctx context.Context, root string, targetFs FileSystem, opts
 		userxattr:                   userxattr,
 		minLayerSize:                config.minLayerSize,
 		allowInvalidMountsOnRestart: config.allowInvalidMountsOnRestart,
+		allowIDMap:                  config.allowIDMap,
+		idmapped:                    make(map[string]interface{}),
```

Comment:
As an optimization, should this only allocate memory if allow ID map is configured?

Reply:
This is accessed concurrently, so it needs to be a `sync.Map`